app/imports/ui/layouts/App.jsx

```diff
@@ -45,10 +47,12 @@ const App = () => {
           <Route path="/list" element={<ProtectedRoute><ListStuff /></ProtectedRoute>} />
           <Route path="/add" element={<ProtectedRoute><AddStuff /></ProtectedRoute>} />
           <Route path="/my-profile" element={<StudentProtectedRoute ready={ready}><MyProfile /></StudentProtectedRoute>} />
-          <Route path="/add-event" element={<ProtectedRoute><AddEvent /></ProtectedRoute>} />
+          <Route path="/event/:_id" element={<ProtectedRoute><EventPage /></ProtectedRoute>} />
+          <Route path="/add-event" element={<AdminProtectedRoute ready={ready}><AddEvent /></AdminProtectedRoute>} />
           <Route path="/list-events" element={<ProtectedRoute><ListEvents /></ProtectedRoute>} />
+          <Route path="/events" element={<ProtectedRoute><MainEventsList /></ProtectedRoute>} />
           <Route path="/company-listing" element={<ProtectedRoute><CompanyListing /></ProtectedRoute>} />
-          <Route path="/edit/:_id" element={<ProtectedRoute><EditEvent /></ProtectedRoute>} />
+          <Route path="/edit/:_id" element={<AdminProtectedRoute ready={ready}><EditEvent /></AdminProtectedRoute>} />
```

Comment:
`/edit/:id` should probably also be a `CompanyProtectedRoute`, assuming we want companies to be able to add events. We would check if the user is a company, if their id matches the company id of the event. If they are an admin, they can edit whatever they want.

Also, if it is a quick fix, `/edit/:id` should be something like `/edit-event/:id`, otherwise it will conflict with editing other things, like job listings.